src/stavisky_lab_to_nwb/stavisky/staviskyecephysinterface.py

```diff
@@ -1,23 +1,192 @@
 """Class for converting generic ecephys data."""
-from pynwb import NWBFile
+import json
+import redis
+import numpy as np
+from pynwb import NWBFile, TimeSeries
+from typing import Optional, Union, Literal
+from hdmf.backends.hdf5 import H5DataIO
 
 from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.tools.nwb_helpers import get_module
+from neuroconv.datainterfaces.ecephys.baserecordingextractorinterface import BaseRecordingExtractorInterface
 
 
 class StaviskySpikingBandPowerInterface(BaseDataInterface):
     """Spiking band power interface for Stavisky Redis conversion"""
 
-    def __init__(self):
-        # This should load the data lazily and prepare variables you need
-        pass
+    # TODO: if we decide timestamps need to be aligned with sorting, etc. then
+    # may need to subclass from BaseTemporalAlignmentInterface instead
 
-    def get_metadata(self):
-        # Automatically retrieve as much metadata as possible
-        metadata = super().get_metadata()
+    def __init__(
+        self,
+        port: int,
+        host: str,
+    ):
+        super().__init__(port=port, host=host)
 
-        return metadata
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        chunk_size: int = 1000,
```

Comment:
I'd document this argument somehow - is it in bytes? MB? KiB?

It might be overkill but another option might be to define a specific [GenericDataChunkIterator](https://hdmf.readthedocs.io/en/stable/tutorials/plot_generic_data_chunk_tutorial.html#genericdatachunkiterator-tutorial) for it, which has a similar argument `chunk_mb`, and would turn this conversion option into a `iterator_options` for consistency with other NeuroConv interfaces